apps/files/src/components/FileLinkForm.vue

```diff
@@ -59,7 +59,10 @@
       </div>
       <div class="uk-width-1-1 uk-width-3-5@m">
         <label class="uk-form-label" for=""><span v-translate>Password</span><em v-if="$_passwordEnforced" class="uk-margin-small-left">(<span v-translate>required</span>)</em></label>
-        <input type="password" autocomplete="new-password" class="uk-input" :class="{ 'uk-form-danger': !$_passwordIsValid }" v-model="password" :placeholder="hasPassword && password === null? '********' : placeholder.password" id="oc-files-file-link-password" />
+        <div class="uk-position-relative">
+          <input type="password" autocomplete="new-password" class="uk-input" :class="{ 'uk-form-danger': !$_passwordIsValid }" v-model="password" :placeholder="hasPassword && password === null? '********' : placeholder.password" id="oc-files-file-link-password" />
+          <div uk-close v-if="!$_passwordEnforced && hasPassword && password!=''" class="uk-position-small uk-position-center-right oc-cursor-pointer" @click="password=''" :uk-tooltip="$_passwordRemoveText" />
```

Comment:
seems the `_passwordEnforced` condition does not do anything.

I tried enforcing password only for the "Editor" role and the cross was still visible to discard the password.

Maybe the `_passwordEnforced` function is not working ? As I can see it's not using proper bitmasks...